src/Agent.Plugins/PipelineCache/SavePipelineCacheV0.cs

```diff
@@ -32,16 +33,11 @@ public class SavePipelineCacheV0 : PipelineCacheTaskPluginBase
                 return;
             }
 
-            string[] key = keyStr.Split(
-                new[] { '\n' },
-                StringSplitOptions.RemoveEmptyEntries
-            );
             PipelineCacheServer server = new PipelineCacheServer();
             await server.UploadAsync(
                 context,
-                key, 
+                fingerprints.First(), 
```

Comment:
Since calculating a fingerprints is potentially time-consuming (and can throw an error) you should only calculate the fingerprint for the primary `key` on "save" (and not for all the restore keys which aren't used anyway).